examples/basic/ipc_shm_ipcapi_producer.c

```diff
@@ -0,0 +1,240 @@
+#include <unistd.h>
+
+#include <umf/ipc.h>
+#include <umf/memory_pool.h>
+#include <umf/pools/pool_scalable.h>
+#include <umf/providers/provider_os_memory.h>
+
+#define INET_ADDR "127.0.0.1"
+#define MSG_SIZE 1024
+
+int producer_connect(int port) {
+    struct sockaddr_in consumer_addr;
+    int producer_socket = -1;
+
+    // create a producer socket
+    producer_socket = socket(AF_INET, SOCK_STREAM, 0);
+    if (producer_socket < 0) {
+        fprintf(stderr, "[producer] ERROR: Unable to create socket\n");
+        return -1;
+    }
+
+    fprintf(stderr, "[producer] Socket created\n");
+
+    // set IP address and port the same as for the consumer
+    consumer_addr.sin_family = AF_INET;
+    consumer_addr.sin_port = htons(port);
+    consumer_addr.sin_addr.s_addr = inet_addr(INET_ADDR);
+
+    // send connection request to the consumer
+    if (connect(producer_socket, (struct sockaddr *)&consumer_addr,
+                sizeof(consumer_addr)) < 0) {
+        fprintf(stderr,
+                "[producer] ERROR: unable to connect to the consumer\n");
+        goto err_close_producer_socket_connect;
+    }
+
+    fprintf(stderr, "[producer] Connected to the consumer\n");
+
+    return producer_socket; // success
+
+err_close_producer_socket_connect:
+    close(producer_socket);
+
+    return -1;
+}
+
+int main(int argc, char *argv[]) {
+    char consumer_message[MSG_SIZE];
+    int producer_socket;
+    int ret = -1;
+
+    if (argc < 2) {
+        fprintf(stderr, "Usage: %s port\n", argv[0]);
+        return -1;
+    }
+
+    int port = atoi(argv[1]);
+
+    umf_memory_provider_handle_t OS_memory_provider = NULL;
+    umf_os_memory_provider_params_t os_params;
+    enum umf_result_t umf_result;
+
+    os_params = umfOsMemoryProviderParamsDefault();
+    os_params.visibility = UMF_MEM_MAP_SHARED;
+
+    // create OS memory provider
+    umf_result = umfMemoryProviderCreate(umfOsMemoryProviderOps(), &os_params,
+                                         &OS_memory_provider);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr,
+                "[producer] ERROR: creating OS memory provider failed\n");
+        return -1;
+    }
+
+    size_t page_size;
+    umf_result =
+        umfMemoryProviderGetMinPageSize(OS_memory_provider, NULL, &page_size);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr,
+                "[producer] ERROR: getting the minimum page size failed\n");
+        goto err_destroy_OS_memory_provider;
+    }
+
+    umf_memory_pool_handle_t scalable_pool;
+    umf_result = umfPoolCreate(umfScalablePoolOps(), OS_memory_provider, NULL,
+                               0, &scalable_pool);
+    if (umf_result != UMF_RESULT_SUCCESS) {
+        fprintf(stderr,
+                "[producer] ERROR: creating the UMF jemalloc pool failed\n");
+        goto err_destroy_OS_memory_provider;
+    }
+
+    // Make 3 allocations of size: 1 page, 2 pages and 3 pages
+    void *ptr1, *ptr2, *IPC_shared_memory;
```

Comment:
for what reason we need `ptr1` and `ptr2`? It just blurs the attention.

Reply:
It was a test for correctness of `fd_offset` - it is done in the IPC test.
Removed.